xinference/model/embedding/core.py

```
@@ -154,6 +154,17 @@ def to(self, *args, **kwargs):
             "gte" in self._model_spec.model_name.lower()
             and "qwen2" in self._model_spec.model_name.lower()
         ):
+            import torch
+            torch_dtype_str = self._kwargs.get("torch-dtype")
+            if torch_dtype_str is not None:
+                try:
+                    torch_dtype = getattr(torch, torch_dtype_str)
+                    if torch_dtype not in [torch.float16, torch.float32]:
+                        raise ValueError(f"Unsupported torch dtype: {torch_dtype_str}")
+                except AttributeError:
+                    torch_dtype = torch.float32
```

If user passed a torch not defined, is it better to raise the error?

---

修改使用fp32推理